renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -7570,6 +7571,379 @@ void WrappedVulkan::vkCmdEndRendering(VkCommandBuffer commandBuffer)
   }
 }
 
+template <typename SerialiserType>
+bool WrappedVulkan::Serialise_vkCmdBuildAccelerationStructuresIndirectKHR(
+    SerialiserType &ser, VkCommandBuffer commandBuffer, uint32_t infoCount,
+    const VkAccelerationStructureBuildGeometryInfoKHR *pInfos,
+    const VkDeviceAddress *pIndirectDeviceAddresses, const uint32_t *pIndirectStrides,
+    const uint32_t *const *ppMaxPrimitiveCounts)
+{
+  SERIALISE_ELEMENT(commandBuffer);
+  SERIALISE_ELEMENT(infoCount).Important();
+  SERIALISE_ELEMENT_ARRAY(pInfos, infoCount);
+  SERIALISE_ELEMENT_ARRAY(pIndirectDeviceAddresses, infoCount);
+  SERIALISE_ELEMENT_ARRAY(pIndirectStrides, infoCount);
+
+  // Convert the array of arrays for easier serialisation
+  {
+    rdcarray<rdcarray<uint32_t>> maxPrimitives;
+    maxPrimitives.reserve(infoCount);
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+    {
+      maxPrimitives.push_back({});
+      auto &rangeInfo = maxPrimitives.back();
+
+      rangeInfo.resize(pInfos[i].geometryCount);
+      memcpy(rangeInfo.data(), ppMaxPrimitiveCounts[i], rangeInfo.size() * sizeof(uint32_t));
+    }
+    SERIALISE_ELEMENT(maxPrimitives);
+  }
+
+  SERIALISE_CHECK_READ_ERRORS();
+
+  if(IsReplayingAndReading())
+  {
+    size_t tempmemSize = sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+    for(uint32_t i = 0; i < infoCount; ++i)
+      tempmemSize += GetNextPatchSize(&pInfos[i]);
+
+    byte *memory = GetTempMemory(tempmemSize);
+    auto *unwrappedInfos = reinterpret_cast<VkAccelerationStructureBuildGeometryInfoKHR *>(memory);
+    memory += sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+      unwrappedInfos[i] = *UnwrapStructAndChain(m_State, memory, &pInfos[i]);
+
+    ObjDisp(commandBuffer)
+        ->CmdBuildAccelerationStructuresIndirectKHR(Unwrap(commandBuffer), infoCount,
+                                                    unwrappedInfos, pIndirectDeviceAddresses,
+                                                    pIndirectStrides, ppMaxPrimitiveCounts);
```

Comment:
`ppMaxPrimitiveCounts` hasn't been serialised here, so on replay it will just be NULL. This will need to be repointed to our local array-of-arrays (I think you'll need to create a temp array of pointers to satisfy the API).